joblib/_memmapping_reducer.py

```diff
@@ -600,12 +607,15 @@ def _cleanup():
                 warnings.warn("Failed to delete temporary folder: {}"
                               .format(pool_subfolder))
 
-        self._finalizers[pool_subfolder] = atexit.register(_cleanup)
+        self._finalizers[context_id] = atexit.register(_cleanup)
 
     def _unlink_temporary_resources(self, context_id=None):
         """Unlink temporary resources created by a process-based pool"""
         if context_id is None:
-            for context_id in self._cached_temp_folders:
+            # iterate over a copy of the cache keys because
+            # unlink_temporary_resources further deletes an entry in this
+            # cache
+            for context_id in list(self._cached_temp_folders):
```

Comment:
Isn't the following safer?

```suggestion
            for context_id in self._cached_temp_folders.copy():
```

Reply:
you're right.